core/input/input.cpp

```diff
@@ -1267,6 +1268,7 @@ Input::JoyEvent Input::_get_mapped_axis_event(const JoyDeviceMapping &mapping, J
 					case TYPE_AXIS:
 						event.index = (int)binding.output.axis.axis;
 						event.value = value;
+						r_range = binding.output.axis.range;
```

Comment:
Isn't the switch below on `axis.range` meant to handle this exact issue already? I'm not deeply familiar with that code but I wonder why this normalisation is handled in two different locations.

Reply:
I'm also not super familiar with what's going on here, but my impression is that the private method is converting to match the `gamecontrollerdb.txt` entry, and `joy_axis` is normalizing consistently for 0.0 to 1.0. On Android and iOS we should be able to bypass this entirely for the triggers. I wasn't sure about the private method conversion so I didn't want to mess with it, so I figured using the binding as the source of truth for the trigger-specific normalization was a reliable way to get it working.

Maybe the private method can branch on if it's an L/R trigger inside the switch and do the proper conversion for those specific axes only once? Because it seems like the switch in `joy_axis` is correcting an error introduced by the switch in `_get_mapped_axis_event` not being specific enough.

Reply:
I've tested this and it works well and makes some sense, but I'm not sure I entirely get what you mean by this comment. Are you trying to say that you're avoiding updating `gamecontrollerdb.txt` and that it isn't configured to the right axis range? Or are you trying to say that the range is correct in `gamecontrollerdb.txt` and that this PR is making sure that Godot converts from those values appropriately (which it wasn't before?)

Reply:
I believe that the `gamecontrollerdb.txt` entry is correct (I recall cross-checking across some other repos), but Godot wasn't correctly converting them specifically in the case of the triggers.

Reply:
In which case then it all looks correct to me. I see that @RandomShaper has already marked it for 4.3 milestone but let's focus on getting this into 4.3/master ASAP (after 4.2 release) as it can be understandably annoying if certain controllers don't work correctly on Android. We could then introduce it as a hotfix if it checks out.

Reply:
Well, we decided to lie on the (maybe too) safe side. If you have tested it and think the changes are good, let's reschedule for 4.2.

Reply:
This seems ok for 4.2 if we merge it now so we get at least a couple betas/RCs of testing.